File: backend/rule-engine/src/main/java/com/agriscope/rule_engine/domain/model/Seed.java
package com.agriscope.rule_engine.domain.model;

import com.agriscope.rule_engine.domain.enums.SeedType;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Seed {
    private SeedType seedType;
    private String displayName;
    private String scientificName;

    private Double minTemperature;
    private Double maxTemperature;

    private Double minSoilMoisture;
    private Double waterRequirement;

    private Double frostRiskTemperature;
    private Double heatStressTemperature;
    private Double heavyRainThreshold;
    private Double seedCoefficient;
    private Double allowedWaterDeficit;
}

File: frontend/src/app/models/Seed.ts
import { SeedType } from "./SeedType";

export interface Seed {
  id: string;
  seedType: SeedType;
  displayName: string;
  scientificName: string;

  minTemperature: number;
  optimalTemperature: number;
  maxTemperature: number;

  minSoilMoisture: number;
  optimalSoilMoisture: number;
  maxSoilMoisture: number;

  waterRequirement: number;

  frostRiskTemperature: number;
  heatStressTemperature: number;
  heavyRainThreshold: number;

  icon?: string; 
}
